cool_seq_tool/data_sources/feature_overlap.py

```diff
@@ -0,0 +1,180 @@
+            `type`, `chromosome` (chromosome without 'chr' prefix), `cds_start`,
+            `cds_stop`, `info_name` (name of record), and `gene`
+        """
+        df = pd.read_csv(
+            self.mane_refseq_gff_path,
+            sep="\t",
+            header=None,
+            skiprows=9,
+            usecols=[0, 2, 3, 4, 8],
+        )
+        df.columns = ["chromosome", "type", "cds_start", "cds_stop", "info"]
+
+        # Restrict to only feature of interest: CDS (which has gene info)
+        df = df[df["type"] == "CDS"].copy()
+
+        # Get name from the info field
+        df["info_name"] = df["info"].apply(
+            lambda info: re.findall("Name=([^;]+)", info)[0]
+        )
+
+        # Get gene from the info field
+        df["gene"] = df["info"].apply(lambda info: re.findall("gene=([^;]+)", info)[0])
+
+        # Get chromosome names without prefix and without suffix for alternate
+        # transcripts
+        df["chromosome"] = df["chromosome"].apply(
+            lambda chromosome: chromosome.strip("chr").split("_")[0]
+        )
+        df["chromosome"] = df["chromosome"].astype(str)
+
+        # Convert start and stop to ints
+        df["cds_start"] = df["cds_start"].astype(int)
+        df["cds_stop"] = df["cds_stop"].astype(int)
+
+        # Only retain certain columns
+        df = df[
+            ["type", "chromosome", "cds_start", "cds_stop", "info_name", "gene"]
+        ]
+
+        return df
+
+    def _get_chr_from_alt_ac(self, identifier: str) -> str:
+        """Get chromosome given genomic identifier
+
+        :param identifier: Genomic identifier on GRCh38 assembly
+        :raises FeatureOverlapError: If unable to find associated GRCh38 chromosome
+        :return: Chromosome. 1..22, X, Y. No 'chr' prefix.
+        """
+        aliases, error_msg = self.seqrepo_access.translate_identifier(
+            identifier, "GRCh38"
+        )
+
+        if error_msg:
+            raise FeatureOverlapError(str(error_msg))
+
+        if not aliases:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 aliases for: {identifier}"
+            )
+
+        chr_pattern = r"^GRCh38:(?P<chromosome>X|Y|([1-9]|1[0-9]|2[0-2]))$"
+        for a in aliases:
+            chr_match = re.match(chr_pattern, a)
+            if chr_match:
+                break
+
+        if not chr_match:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 chromosome for: {identifier}"
+            )
+
+        chr_groupdict = chr_match.groupdict()
+        return chr_groupdict["chromosome"]
+
+    def get_grch38_cds_overlap(
+        self,
+        start: int,
+        end: int,
+        chromosome: Optional[str] = None,
+        identifier: Optional[str] = None,
+        residue_mode: ResidueMode = ResidueMode.RESIDUE,
+    ) -> Optional[Dict]:
+        """Get feature overlap for GRCh38 genomic data
```

Comment:
Would someone (Kori? @austinant?) be willing -- for my sake -- to add a descriptive sentence or two here? I can also paraphrase something from the manuscript if there's a specific relevant section

Reply:
@jsstevenson updated in latest commit. Let me know if that clears things up for you

Reply:
@austinant how about

```python
        """Given GRCh38 genomic data, find the overlapping MANE features (gene and cds).
        The genomic data is specified as a sequence location by `chromosome`, `start`,
        `end`. All CDS regions with which the input sequence location has nonzero base
        pair overlap will be returned.

        :return: MANE feature (gene/cds) overlap data represented as a dict. The
            dictionary will be keyed by genes which overlap the input sequence location.
            Each gene contains a list of the overlapping CDS regions with the beginning
            and end of the input sequence location's overlap with each
            {
                gene: {
                    'cds': VRS Sequence Location
                    'overlap': VRS Sequence Location
                }
            }
        """
```